app/controllers/chargeback_controller.rb

```diff
@@ -873,4 +1003,34 @@ def set_session_data
     session[:chargeback_current_page] = @current_page
     session[:chageback_display]       = @display unless @display.nil?
   end
+
+  def display_detail_errors(detail, errors)
+    errors.each { |field, msg| add_flash("'#{detail.description}' #{field.to_s.capitalize} #{msg}", :error) }
+  end
+
+  def add_row(i, pos)
+    render :update do |page|
+      page.replace("rate_detail_row_#{i}_0", :partial => "tier_first_row") # Update the first row to change the colspan
+      page.insert_html(:after, "rate_detail_row_#{i}_#{pos}", :partial => "tier_row") # Insert the new tier after the last one
+      page << javascript_for_miq_button_visibility(true)
+    end
+  end
+
+  def replace_rows(detail_index, tiers, tier_to_remove_index)
+    render :update do |page|
+      page.replace("rate_detail_row_#{detail_index}_0", :partial => "tier_first_row")
+      tiers.each_with_index do |_tier, tier_index|
+        next if tier_index <= tier_to_remove_index
+        # Move up tiers not to have blank rows
+        # @edit[:new][:tiers][detail_index][tier_index - 1] = @edit[:new][:tiers][detail_index][tier_index]
+        params[:tier_row] = tier_index
+        page.replace("rate_detail_row_#{detail_index}_#{tier_index - 1}", :partial => "tier_row")
+        params[:tier_row] = nil
+      end
+      # Delete the last row
+      # delete_row(detail_index, @sb[:num_tiers][detail_index])
```

Comment:
Same, I don't understand why this is here and commented out?  Can we fix this in a followup instead of adding commented code?